networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/LibP2PPrivateKeyLoader.java

```diff
@@ -37,11 +39,22 @@ public LibP2PPrivateKeyLoader(
 
   @Override
   public PrivKey get() {
-    final Bytes privKeyBytes =
-        privateKeySource
-            .map(PrivateKeySource::getOrGeneratePrivateKeyBytes)
-            .orElseGet(() -> generateAndSaveNewPrivateKey(keyValueStore));
-    return KeyKt.unmarshalPrivateKey(privKeyBytes.toArrayUnsafe());
+    final Bytes privKeyBytes;
+    final Optional<PrivateKeySource.Type> type;
+    if (privateKeySource.isEmpty()) {
+      privKeyBytes = generateAndSaveNewPrivateKey(keyValueStore);
```

Comment:
seems like this method should be named getOrGenerateAndSaveNewPrivateKey

Reply:
nice catch!